crypto/src/hash/types.rs

```diff
@@ -31,10 +32,28 @@ impl From<HashBytes> for BaseHash {
     }
 }
 
+impl<'a> From<&'a [u8]> for BaseHash {
+    fn from(b: &'a [u8]) -> Self {
+        let mut o = BaseHash::default();
+        o.0.copy_from_slice(&b);
+        o
+    }
+}
+
+impl BaseHash {
+    pub fn to_hex(&self) -> String {
+        let mut s = String::from("0x");
+        for b in &self.0 {
+            write!(&mut s, "{:02x}", b).expect("failed to write hex byte")
+        }
+        s
+    }
+}
+
```

Comment:
```suggestion
impl AsRef<[u8]> for BaseHash {
    fn as_ref(&self) -> &[u8] {
        &self.0
    }
}
```

and then if we use the `hex` crate this type will automatically implement `to_hex` whenever the trait is in scope!

Reply:
oh nice, so by implementing `AsRef` for a type, then  `to_hex` automatically gets implemented?

Reply:
Yep you got it As long as the trait is in scope that is